### ISSUE TITLE
[JENKINS-46922] Option to disable the changelog while getting jenkinsfile from git

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/git-plugin#3193]
<p>It taken a lot of time to get revision list from git while getting jenkinsfile<br/>
git rev-list e4b4b57f7d79b19f11acdaef7a5a50866d9d763b # timeout=10<br/>
I would be nice to have option to avoid changelog creation during getting jenkinsfile from git (like on checkout operation)</p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=slavik334">slavik334</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-46922" target="_blank">Option to disable the changelog while getting jenkinsfile from git</a></i></summary>
<i><ul>
<li><b>status</b>: Open
<li><b>priority</b>: Major
<li><b>component(s)</b>: git-plugin
<li><b>resolution</b>: Unresolved
<li><b>votes</b>: 11
<li><b>watchers</b>: 5
<li><b>imported</b>: 2025-12-02
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>It taken a lot of time to get revision list from git while getting jenkinsfile
git rev-list e4b4b57f7d79b19f11acdaef7a5a50866d9d763b # timeout=10
I would be nice to have option to avoid changelog creation during getting jenkinsfile from git (like on checkout operation)</p></pre>
</details>
</details>
<details><summary><i>environment</i></summary>

```
Jenkins 2.60.1<br/>
Git plugin 3.4.1
```
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-46922?focusedId=315124&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-315124">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Please describe the cases where you see it taking a long time to get the revision list from git.  Listing revisions is a local operation which should be quite fast.</p>
</pre>
</details>

</ul>
</details>

<p>Please describe the cases where you see it taking a long time to get the revision list from git.  Listing revisions is a local operation which should be quite fast.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>slavik334:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-46922?focusedId=315141&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-315141">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I have a CI test and subtests running in parallel. So every subtest is a separate job. Every job during start gets pipeline from git. If there are several tests with subtests start running - they are trying to get pipeline scripts from git on Master node. And every time jenkins is trying to create changelog during git checkout. The CPU is on top at that moment.</p>

<p>I do not need git rev-list during jenkinsfile checkout and it will save Jenkins master resources. So it will be critically nice to have an option to disable changelog creation as we have it on checkout command in pipeline.</p>

<p>thanks.</p>
</pre>
</details>

</ul>
</details>

<p>I have a CI test and subtests running in parallel. So every subtest is a separate job. Every job during start gets pipeline from git. If there are several tests with subtests start running - they are trying to get pipeline scripts from git on Master node. And every time jenkins is trying to create changelog during git checkout. The CPU is on top at that moment.</p>

<p>I do not need git rev-list during jenkinsfile checkout and it will save Jenkins master resources. So it will be critically nice to have an option to disable changelog creation as we have it on checkout command in pipeline.</p>

<p>thanks.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-46922?focusedId=320149&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-320149">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Git client plugin 2.6.1-SNAPSHOT now includes a <tt>git rev-list</tt> command which has the option to not generate the entire rev-list.  Can you check a <a href="https://ci.jenkins.io/blue/organizations/jenkins/Plugins%2Fgit-client-plugin/detail/master/167/artifacts" class="external-link" target="_blank" rel="nofollow noopener">build from ci.jenkins.io</a> to see if that resolves your concern?</p>
</pre>
</details>

</ul>
</details>

<p>Git client plugin 2.6.1-SNAPSHOT now includes a <tt>git rev-list</tt> command which has the option to not generate the entire rev-list.  Can you check a <a href="https://ci.jenkins.io/blue/organizations/jenkins/Plugins%2Fgit-client-plugin/detail/master/167/artifacts" class="external-link" target="_blank" rel="nofollow noopener">build from ci.jenkins.io</a> to see if that resolves your concern?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>slavik334:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-46922?focusedId=320210&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-320210">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I have installed 2.6.1 (suggested) version of plugin. Could you please tell me where I can find this option to  not to generate entire rev-list?</p>

<p>thanks</p>
</pre>
</details>

</ul>
</details>

<p>I have installed 2.6.1 (suggested) version of plugin. Could you please tell me where I can find this option to  not to generate entire rev-list?</p>

<p>thanks</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-46922?focusedId=320271&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-320271">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=slavik334" class="user-hover" rel="slavik334">slavik334</a> sorry for my incorrect phrasing.  The 2.6.1-SNAPSHOT includes a rev-list implementation which uses the <tt>--no-walk</tt> parameter to the <tt>git rev-list</tt> command when it detects the history is not required.  No configuration required.  Install the plugin and see if the behavior is measurably different.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=slavik334" class="user-hover" rel="slavik334">slavik334</a> sorry for my incorrect phrasing.  The 2.6.1-SNAPSHOT includes a rev-list implementation which uses the <tt>--no-walk</tt> parameter to the <tt>git rev-list</tt> command when it detects the history is not required.  No configuration required.  Install the plugin and see if the behavior is measurably different.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>slavik334:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-46922?focusedId=320278&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-320278">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>thanks, i will try</p>

<p>but seems that preferred way - is to have an option</p>

<p>cause other people may need the option to get changes....</p>
</pre>
</details>

</ul>
</details>

<p>thanks, i will try</p>

<p>but seems that preferred way - is to have an option</p>

<p>cause other people may need the option to get changes....</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-46922?focusedId=320296&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-320296">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=slavik334" class="user-hover" rel="slavik334">slavik334</a> thanks for trying it.  I pointed you to that existing code because I am unlikely to ever implement this request.  There are over 120000 installations of the git plugin worldwide.  I hesitate to add new options unless I'm persuaded there will be many users for that option.</p>

<p>Unless this request has many others declaring "me too", and the request is followed by a pull request implementing it (and implementing tests to confirm behavior), it won't be implemented.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=slavik334" class="user-hover" rel="slavik334">slavik334</a> thanks for trying it.  I pointed you to that existing code because I am unlikely to ever implement this request.  There are over 120000 installations of the git plugin worldwide.  I hesitate to add new options unless I'm persuaded there will be many users for that option.</p>

<p>Unless this request has many others declaring "me too", and the request is followed by a pull request implementing it (and implementing tests to confirm behavior), it won't be implemented.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>slavik334:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-46922?focusedId=321351&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-321351">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a>,</p>

<p>after tests found that change log does not appear twice</p>

<p>seems that the option could be implemented to choose if to disable the changelog</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a>,</p>

<p>after tests found that change log does not appear twice</p>

<p>seems that the option could be implemented to choose if to disable the changelog</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/canuck1987.svg" title="canuck1987's avatar" /> canuck1987:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-46922?focusedId=403477&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-403477">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I would be interested in this as well. We are doing Pre-Commit builds of a master repo, patches onto a different repo. We have a parent MultiBranch Pipeline which triggers many downstream Pipeline jobs. This means the changelog is fairly useless because the branch keeps changing (meaning we get hundreds/thousands of changelog messages).</p>
</pre>
</details>

</ul>
</details>

<p>I would be interested in this as well. We are doing Pre-Commit builds of a master repo, patches onto a different repo. We have a parent MultiBranch Pipeline which triggers many downstream Pipeline jobs. This means the changelog is fairly useless because the branch keeps changing (meaning we get hundreds/thousands of changelog messages).</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>levnagdimunov0:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-46922?focusedId=415321&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-415321">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>We too would benefit from this feature. In our case the repo containing the Jenkinsfile is a helper repo, thus we do not want to include it in the changelog.</p>
</pre>
</details>

</ul>
</details>

<p>We too would benefit from this feature. In our case the repo containing the Jenkinsfile is a helper repo, thus we do not want to include it in the changelog.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>czguang:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-46922?focusedId=437958&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-437958">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>me too.</p>

<p>We develop android apps managed by repo tools, and only care about the change logs of these apps.</p>

<p>Hope there is an option to disable change logs of git repository which manages pipeline script.</p>

<p>Thanks a lot.</p>
</pre>
</details>

</ul>
</details>

<p>me too.</p>

<p>We develop android apps managed by repo tools, and only care about the change logs of these apps.</p>

<p>Hope there is an option to disable change logs of git repository which manages pipeline script.</p>

<p>Thanks a lot.</p>
